src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -82,8 +82,9 @@ public Boolean upsertNode(Node node) throws Exception {
 
     @Override
     public Boolean upsertRelation(Node nodeFrom, Node nodeTo, Map<String, String> relationProperties) throws Exception {
-        try (Session session = neo4jDriver.session()) {
-            try (Transaction transaction = session.beginTransaction()) {
+        Session session = neo4jDriver.session();
+        Transaction transaction = session.beginTransaction();
+            try {
```

Comment:
The previous one is good... new one does the same with extra lines...
only optimisation is -- instead of two try blocks we can have single one...